alacritty/src/input.rs

```diff
@@ -135,6 +140,85 @@ impl<T: EventListener> Execute<T> for Action {
                     Err(err) => warn!("Couldn't run command {}", err),
                 }
             },
+            Action::ToggleKeyboardMode => ctx.terminal_mut().toggle_keyboard_mode(),
```

Comment:
I feel like the style of the cursor really needs to change in this mode. Currently there's no indication of what mode the user is currently in, and making that indication the cursor itself makes a lot of sense to me. However, the style of the cursor is already user configurable, which makes this more difficult.

I could imagine something like a bright colored border around the cursor (block or otherwise), but I'll admit there's got to be a better solution.

Reply:
Also, I think it's strange that some keyboard actions still pass through while in this mode. For example, I find it strange that the arrow keys continue to work as my shell's history. Personally I think the arrow keys should be wired up to cursor movement in keyboard mode, but that's another point.

Reply:
> I could imagine something like a bright colored border around the cursor (block or otherwise), but I'll admit there's got to be a better solution.

I'm not a fan of this solution at all, I don't think it provides much value but will likely ruin this mode completely for many people (myself included).

I've proposed to just make the keyboard motion cursor configurable, similar to the normal cursor, so that would allow using block in this mode but underline normally for example.

I'm not sure how far we should go though for example. Do we need to allow just different cursor shapes, or do we also need different cursor colors?

Reply:
> For example, I find it strange that the arrow keys continue to work as my shell's history. Personally I think the arrow keys should be wired up to cursor movement in keyboard mode, but that's another point.

I've considered this but wasn't sure if I should make this the default. We could in theory add `~TermMode::KEYBOARD_MOTION` to all existing bindings that send characters to the terminal.

Reply:
> I've proposed to just make the keyboard motion cursor configurable, similar to the normal cursor, so that would allow using block in this mode but underline normally for example.
>
> I'm not sure how far we should go though for example. Do we need to allow just different cursor shapes, or do we also need different cursor colors?

I frankly don't think the existing styles for the cursor are appropriate to indicate this mode. I suggested some kind of border specifically because a colored cursor is hard since we already color the cursor based on the text color it's over, and none of our styles for the cursor: (▇) Block, (_) Underline, or (|) Beam indicate a change in mode in my opinion.

> I've considered this but wasn't sure if I should make this the default. We could in theory add `~TermMode::KEYBOARD_MOTION` to all existing bindings that send characters to the terminal.

I'm just not sure what's the right way to go. I'd expect bindings like `ToggleFullscreen` or `ClearLogNotice` to work, while bindings like a bare arrow key or character don't pass through. I think it's a somewhat interesting question of what to do with actions like `ScrollPageUp`, `ClearHistory`, and `Paste` (also, what about the existing `Copy` binding?).

Reply:
> I'm just not sure what's the right way to go. I'd expect bindings like ToggleFullscreen or ClearLogNotice to work, while bindings like a bare arrow key or character don't pass through. I think it's a somewhat interesting question of what to do with actions like ScrollPageUp, ClearHistory, and Paste (also, what about the existing Copy binding?).

I think you might be making this more complicated than it actually is. I feel like there's a trivial solution for this. Bindings that interact with Alacritty should work, while bindings that interact with the PTY should not. So we just have to change all bindings that use `Action::Esc` to be disabled in keyboard motion mode.